infra/aws/govcloud.md

```diff
@@ -0,0 +1,199 @@
+# Working in GovCloud
+
+Many of our government clients will want to have at least their production
+infrastructure in [AWS GovCloud](https://aws.amazon.com/govcloud-us/), which
+is a set of AWS regions that are separate from commercial AWS regions, with
+a number of additional controls that allow them to conform to the higher
+security and compliance standards needed to meet federal guidelines like
+[FedRAMP](https://www.fedramp.gov/).
+
+## Notable Considerations With GovCloud
+
+* Every GovCloud account must be attached to a corresponding account in
+  commercial AWS (mostly for billing purposes). This has some impact on
+  how you build AWS Organizations in GovCloud (see below).
+* Commercial AWS and GovCloud resources cannot interact directly in any
+  way. This means that you cannot assume roles between the two or tie
+  a commercial resource to a GovCloud resource unless using some publicly
+  addressable method.
+* GovCloud resources use a different AWS partition; this is the second
+  part of an ARN. So while an ARN for a commercial resource might look
+  like this:
+
+  ```text
+  arn:aws:iam::123456789000:user/testuser
+  ```
+
+  a GovCloud ARN will look like this:
+
+  ```text
+  arn:aws-us-gov:iam::123456789000:user/testuser
+  ```
+
+  This means that any Terraform modules will need to take this into
+  account; see the Terraform
+  [`aws_partition`](https://www.terraform.io/docs/providers/aws/d/partition.html)
+  data source for how to do this.
+* Many AWS services are not available in GovCloud; some are available
+  in one GovCloud region and not others. See the GovCloud
+  [Services in Scope](https://aws.amazon.com/compliance/services-in-scope/)
+  and the AWS [Region
+  Table](https://aws.amazon.com/about-aws/global-infrastructure/regional-product-services/)
+  for more information.
+* Cloudtrail is automatically added to all accounts in GovCloud by default.
+  This means that you will have to handle things differently when trying to
+  configure an organization's Cloudtrail.
+
+## Setting Up a GovCloud Organization
+
+### Create Your GovCloud Organization Master Account
+
+To set up a new GovCloud AWS Organization, you must log in as root to the
```

Comment:
Do you use the existing org-root AWS account for the rest of commercial or do create a new master commercial AWS account which will be org-root-gov or something?

Reply:
I don't have an Organization set up for my AWS account (or maybe the root account is). So does that change things? Or would you suggest starting with AWS Org setup before starting GovCloud work?

Reply:
@dynamike I used the existing org-root account when I did this; doing that will keep all your billing tied to that same account, which makes things simpler, and there's no real reason to create a whole new account just to spin up your GovCloud organization.

@chrisgilmerproj I would recommend setting up an organization for your commercial account(s) first before starting off with GovCloud. That allows you to use the `aws organizations` commands to create the GovCloud accounts (since you can't do that without an organization, I assume). You could always create a new account (milmove-org-root or whatever), create your organization there, and then move your existing account into the new organization. That will let you put off your migration of AWS resources into an organization-centric arrangement in commercial, but still be able to tie your GovCloud account to it and do GovCloud in an organization-centric fashion.